jdaviz/configs/default/plugins/metadata_viewer/metadata_viewer.py

```diff
@@ -47,7 +47,11 @@ def __init__(self, *args, **kwargs):
 
     @property
     def user_api(self):
-        return PluginUserApi(self, expose=('dataset', 'show_primary'), readonly=('metadata',))
+        plg_api = PluginUserApi(self, expose=('dataset', 'show_primary'),
+                                readonly=('metadata', 'meta',))
+        plg_api._deprecation_msg = ("in the future, \"metadata\" will not be publicly accessible, "

```

Comment:
I don't think this is correct but I cannot find a good example of deprecating just an attribute that is also traitlet. Just so we're clear, we are never going to delete this, just retiring it from *public* API. Maybe @kecnry can tell me how when he returns next week.

Reply:
right, this is intended when the entire plugin (or plugin name) is deprecated.  I don't know if we've had a case with an attribute before, so we can either write logic into `PluginUserAPI` (maybe a separate `deprecated` list that acts similar to `readonly` but raising the warnings), or just make `meta` a property here and use the deprecated decorator.

Reply:
OK I reverted this and added something for attr in the API parent class.